src/asyncRaw.c

```diff
@@ -499,7 +499,15 @@ static size_t asyncHeaderCallback(char* buffer, size_t size, size_t nitems, void
         free(tmp);
         return nitems * size;
     }
-    value = strtok_r(NULL, ":", &save);
+    // Added support for handling Location header with absolute path
+    if (strstr(name, "Location") && (strstr(save, "http://") || strstr(save, "https://")))
```

Comment:
You really shouldn't be hitting save directly. That's not guaranteed to be the rest of the string, it just happens to be in some implementations. The right solution for this would be to change the : to something like \r which will be at the end of the string. That way we don't have to treat Location specially.